### PR TITLE
OpenVINO EP Rel 1.18 Changes

### PR DESCRIPTION
**Pull request by sfatimar**
### Description
These changes include
Support to OpenVINO 2024.1 
Import PreCompiled Blobs with EPContext Blob 
Separate Device/Precision as input
Deprecate CPU_FP32 , GPU_FP32 terminology , introduce CPU, GPU 
AUTO GPU, CPU will only create GPU Blob and not CPU Blob. 

### Motivation and Context
- OpenVINO 2024.1 will be out soon
- Import Precompiled Blob can greatly reduce FEIL/FIL Time. 
- Separating Device/Precision will make the input cleaner
-

### CODE REVIEW

**Comment by sfatimar** [comment]
@jywu-msft

**Comment by jywu-msft** [comment]
lint caught a couple typos, 
also python lint/format required pipeline is failing. please take a look.

**Comment by jywu-msft** [comment]
this test hit exception and was aborted. 
1: [----------] 1 test from ParallelExecutor
1: [ RUN      ] ParallelExecutor.TestStatusPropagation
1/7 Test #1: onnxruntime_test_all ....................Subprocess aborted***Exception: 279.98 sec

**Comment by sspintel** [comment]
@jywu-msft  Please help re-run pipelines.

**Comment by sspintel** [comment]
@jywu-msft Fixed the pipeline failure that happened due to onnx backend test filters. Kindly help re-run.

**Comment by sspintel** [comment]
@jywu-msft We had to pull in a critical fix so the pipelines need to be restarted (earlier macOS pipeline was failing due to an unrelated reason)

We don't expect lint failures, and so can you please help run all pipelines?

**Comment by preetha-intel** [comment]
@jywu-msft There was a bug fix for one of the feature wrt NPU device. We have fixed it. Kindly help re-run the pipelines. We have fixed the lint and unit test cases failures. Anticipating no more changes to this PR.

**Comment by jywu-msft** [comment]
> @jywu-msft There was a bug fix for one of the feature wrt NPU device. We have fixed it. Kindly help re-run the pipelines. We have fixed the lint and unit test cases failures. Anticipating no more changes to this PR.

can you mark it ready for review (if no more expected code changes)
I'll kick off the pipelines

**Comment by sfatimar** [comment]
@jywu-msft PR is ready for review

File: onnxruntime/test/perftest/ort_test_session.cc [hunk 1]
```diff
@@ -304,6 +344,15 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
         } else {
           ov_options[key] = value;
         }
+      } else if (key == "export_ep_ctx_blob") {
```
Comment (jywu-msft) [inline onnxruntime/test/perftest/ort_test_session.cc:347]: can we use the existing session options
ep.context_enable, ep.context_embed_mode and ep.context_file_path ?

**Comment by jywu-msft** [comment]
MacOS failure is under investigation. (unrelated to this PR)

**Comment by jywu-msft** [comment]
can you merge latest main?

**Comment by sspintel** [comment]
@jywu-msft Merged latest main.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**